lib/taurus/core/tango/tangoattribute.py

```diff
@@ -135,12 +139,21 @@ def __init__(self, attr=None, pytango_dev_attr=None, config=None):
 
     def __getattr__(self, name):
         try:
-            ret = getattr(self._attrRef, name)
+            try:
+                # Use the attr reference if the attr is still valid
+                ret = getattr(self._attrRef, name)
+            except ReferenceError:
+                # re-create the attribute in case it was no longer referenced
+                # (this may happen in some rare cases in which the value is
+                # still used but the attr is no longer referenced elsewhere)
+                from taurus import Attribute
```

Comment:
You can use the tango factory Instead of use the Taurus Helper API to create the Attribute
f = TangoAttribute.factory()
a = f.getAttribute(self.__attrName)

Reply:
Good catch. Changed.